### PR TITLE
Bug 1845530 - Update AndroidX Paging to version 3.2.0

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1845530

### CODE REVIEW

**Comment by rvandermeulen** [comment]
FYI, version 3.2.0 was released today.
https://developer.android.com/jetpack/androidx/releases/paging#3.2.0

File: android-components/plugins/dependencies/src/main/java/DependenciesPlugin.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ object Versions {
         const val espresso = "3.5.1"
         const val room = "2.4.3"
         const val savedstate = "1.2.1"
-        const val paging = "2.1.2"
+        const val paging = "3.1.1"
```
Comment (rvandermeulen) [inline android-components/plugins/dependencies/src/main/java/DependenciesPlugin.kt:80]: Can we remove the duplication from FenixDependencies now too?

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt [hunk 2]
```diff
@@ -105,7 +105,7 @@ class GeckoSitePermissionsStorage(
 
         return onDiskStorage.getSitePermissionsPaged().map { onDiskPermission ->
             val geckoPermissions = geckoPermissionsByHost[onDiskPermission.origin].groupByType()
-            mergePermissions(onDiskPermission, geckoPermissions)
+            mergePermissions(onDiskPermission, geckoPermissions) ?: onDiskPermission
```
Comment (mcarare) [inline android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:108]: @Amejia481 The alternative here was to use the not-null assertion operator. Is this mapping still correct?

Comment (Amejia481) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:108]: Could you help to clarify why do we need to provide a not-null value?

Comment (mcarare) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:108]: The signature for the map method in `DataSource` class has changed in the new library version, and the code was migrated to Kotlin. It now expects a non-null result for mapping function, and the mergePermissions method returns nullable `SitePermissions`

Comment (Amejia481) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:108]: Yes, I think it's safe to add the Elvis operator!

**Review by rvandermeulen — APPROVED** [review]
Dependencies changes look good to me. I'll defer to Arturo for the code changes, however 😄

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**